app/models/admin.rb

```diff
@@ -15,4 +18,21 @@ class Admin < ApplicationRecord
   validates :email, uniqueness: true
   validates_presence_of :password_confirmation, if: :new_record?
   validates_confirmation_of :password, if: :new_record?
+
+  validate :allowed_roles
+  def allowed_roles
+    if roles && (roles - VALID_ROLES).present?
+      errors.add(:roles, :invalid)
+    end
+  end
+
+  ### PERMISSIONS ###
+
+  def admin_post_access?
+    AdminPostPolicy.can_post?(self)
```

Comment:
Does `policy(AdminPost).can_post?` work here? If so, we can remove `AdminPostPolicy.can_post?`.

Reply:
It appears that the "policy" method is only available in views and controllers, so we can't call that here.

Reply:
Gotcha. In that case, can we use `policy` in the views, and remove the `### PERMISSIONS ###` methods here?